### ISSUE TITLE
v2.23 complains of invalid gaphor model in file that opens in v2.22

### ISSUE THREAD

**Issue by mikekidner** [gaphor/gaphor#2995]
### Describe the bug
A (large) model that opens in v2.22 does not open in v2.23.  

Gaphor version:         2.23.0
Operating System:       Darwin (21.6.0)
Display:                GdkMacosDisplay
Python version:         3.11.7
GTK version:            4.12.4
Adwaita version:        1.4.2
GtkSourceView version:  5.10.0
Cairo version:          1.18.0
Pango version:          1.50.14
PyGObject version:      3.46.0
Pycairo version:        1.25.1
pygit2/libgit2 version: 1.13.3  / 1.7.1

Errors:

Time since application startup: 0:00:55
|Traceback (most recent call last):
|  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/ui/errorhandler.pyc", line 42, in response
|  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/ui/filemanager.pyc", line 234, in <lambda>
|  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/core/eventmanager.pyc", line 83, in handle
|  File "/Applications/Gaphor.app/Contents/Frameworks/generic/event.pyc", line 61, in handle
|ExceptionGroup: Error while handling events (1 sub-exception)
└─┬──────────────────────────────╌┄┈
  |Traceback (most recent call last):
  |  File "/Applications/Gaphor.app/Contents/Frameworks/generic/event.pyc", line 57, in handle
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/application.pyc", line 119, in on_session_shutdown
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/application.pyc", line 147, in shutdown_session
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/application.pyc", line 231, in shutdown
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/application.pyc", line 238, in shutdown_service
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/core/modeling/elementfactory.pyc", line 76, in shutdown
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/core/modeling/elementfactory.pyc", line 197, in flush
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/core/modeling/diagram.pyc", line 362, in unlink
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphas/connections.pyc", line 165, in remove_connections_to_item
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphas/connections.pyc", line 148, in _disconnect_item
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/diagram/_connector.pyc", line 108, in __call__
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/UML/classes/containmentconnect.pyc", line 68, in disconnect
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/core/modeling/element.pyc", line 213, in __setattr__
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/core/modeling/properties.pyc", line 149, in __set__
  |  File "/Applications/Gaphor.app/Contents/Frameworks/gaphor/core/modeling/properties.pyc", line 381, in set
  |TypeError: Can not set <gaphor.UML.uml.Package element b0d8bd42-2066-11ee-b860-ba28606445aa>.package to itself
  └─────────────────────────────╌┄┈

### Expected behavior
The model to open. Or to identify the issue, so it can be corrected.

### OS
- [ ] Linux (Please put in notes the specific distro)
- [x ] macOS
- [ ] Windows

### Version
Version of Gaphor:2.23

### Additional information
The model opens in 2.22, is there a way to identify the problematic elements?

**Comment by danyeaw** [comment]
Hey @mikekidner, is it possible to share the model?

**Comment by mikekidner** [comment]
Not publicly, but is there a way I can upload you a copy?

**Comment by amolenaar** [comment]
I'm surprised it does open with 2.22. This check is in there for quite some time. An element should not reference itself in the model.

In the model file, there should be a package with `id="b0d8bd42-2066-11ee-b860-ba28606445aa"`, that contains 

```
<package>
<ref refid="b0d8bd42-2066-11ee-b860-ba28606445aa"/>
</package>
```

If you remove the package tag with contents, the model should load again.

**Comment by mikekidner** [comment]
@danyeaw mikekidneratmacdotcom

**Comment by tompkins-ct** [comment]
Hi,

I'm getting the same issue with a model I have, unfortunately, I can share the model. It opens in 2.22.1 but fails to open in 2.23.1 with the same error "This file does not contain a valid Gaphor model". 

I'm not sure how to see the error log because the window closes as soon as I close the error dialog. 

Running via the cmd line gives the following (but that may just be from starting the debug session):
As far as I can tell no exceptions popped up.

```. 
gaphor.storage.storage INFO Read 5537 elements from file
> /Users/tompkins/.pyenv/versions/3.11.5/lib/python3.11/site-packages/gaphor/core/modeling/diagram.py(95)rgetattr()
-> v = getattr(obj, attrname(obj, name), NO_ATTR)
```

**Comment by danyeaw** [comment]
Thanks @mikekidner for the model and @tompkins-ct for the additional traceback, both of those were very helpful to debug this. I opened https://github.com/gaphor/gaphor/pull/3018 to resolve this. :+1:

**Comment by danyeaw** [comment]
Closed by #3018.

**Comment by mikekidner** [comment]
Thanks @danyeaw! It's a relief to be able to open that model again.
